src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -658,6 +661,71 @@ bool OutputStateMachineEngine::_IntermediateSpaceDispatch(const wchar_t wchActio
     return fSuccess;
 }
 
+// Routine Description:
+// - Handles actions that have an intermediate '#' (0x23), such as XTPUSHSGR, XTPOPSGR
+// Arguments:
+// - wch - Character to dispatch.
+// Return Value:
+// - True if handled successfully. False otherwise.
+bool OutputStateMachineEngine::_IntermediateHashDispatch(const wchar_t wchAction,
+                                                         _In_reads_(cParams) const unsigned short* const rgusParams,
+                                                         const unsigned short cParams)
+{
+    bool fSuccess = false;
+
+    DispatchTypes::GraphicsOptions rgPushPopParams[StateMachine::s_cParamsMax];
+    size_t cOptions = StateMachine::s_cParamsMax;
+    // Ensure that there was the right number of params
+    switch (wchAction)
+    {
+    case VTActionCodes::XT_PushSgr:
+    case VTActionCodes::XT_PushSgrAlias:
+        fSuccess = _GetTypedParams(rgusParams, cParams, rgPushPopParams, &cOptions);
+        break;
+
+    case VTActionCodes::XT_PopSgr:
+    case VTActionCodes::XT_PopSgrAlias:
+        if (cParams > 0)
+        {
+            // Can't supply params for XTPOPSGR.
+            fSuccess = false;
+        }
+        else
+        {
+            fSuccess = true;
+        }
+        break;
+
+    default:
+        // If no params to fill, param filling was successful.
+        fSuccess = true;
+        break;
+    }
+    if (fSuccess)
+    {
+        switch (wchAction)
+        {
+        case VTActionCodes::XT_PushSgr:
+        case VTActionCodes::XT_PushSgrAlias:
+            fSuccess = _dispatch->PushGraphicsRendition(rgPushPopParams, cOptions);
+            // TODO: telemetry

```

Comment:
This should either get done, or frankly ignored. I don't think we're actually doing anything with the state machine logging the other cases we have tbh.

Reply:
done